cmake/modules/CheckCompiler.cmake

```diff
@@ -48,14 +48,17 @@ if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
   message(STATUS "Found Clang. Major version ${CLANG_MAJOR}, minor version ${CLANG_MINOR}")
   set(COMPILER_VERSION clang${CLANG_MAJOR}${CLANG_MINOR})
   if(CMAKE_GENERATOR STREQUAL "Ninja")
-    # https://bugzilla.samba.org/show_bug.cgi?id=8118 and color.
     # LLVM/Clang are automatically checking if we are in interactive terminal mode.
     # We use color output only for Ninja, because Ninja by default is buffering the output,
     # so Clang disables colors as it is sure whether the output goes to a file or to a terminal.
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fcolor-diagnostics")
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fcolor-diagnostics")
   endif()
-  if(ccache)
+  if(ccache AND CCACHE_VERSION VERSION_LESS "3.2.0")
+    # https://bugzilla.samba.org/show_bug.cgi?id=8118
+    # Call to 'ccache clang' is triggering next warning (valid for ccache 3.1.x, fixed in 3.2):
+    # "clang: warning: argument unused during compilation: '-c"
+    # Adding -Qunused-arguments provides a workaround for the bug.
```

Comment:
Is it really worth it to keep a workaround for such an old version of ccache? Version 3.2 was released in 2014... https://ccache.samba.org/releasenotes.html#_ccache_3_2

Reply:
Maybe in case if people still using old version, otherwise we can put requirement on ccache version..
Actually I see EPEL is still shipping cache-3.1.6-2.el6.i686.rpm for Centos 6. It will be very rare case, that a person will use i686 + ccache + clang, but it could happen :)

Reply:
2014, just like GCC 4.9, which we also support :-)